aten/src/ATen/native/sparse/SoftMax.cpp

```diff
@@ -531,13 +630,15 @@ Tensor log_softmax_sparse_cpu(const Tensor& input_, const int64_t dim_, const bo
   }
   TORCH_CHECK(dim_ >= 0 && dim_ < input.dim(),
               "dim must be non-negative and less than input dimensions");
-  AT_DISPATCH_FLOATING_TYPES(input.scalar_type(), "log_softmax", [&] {
-      cpu_sparse_coo_softmax<scalar_t, true>(output, input, dim_);
-  });
+  if (input.is_cuda()) {
```

Comment:
I think you want to leave this dispatch logic to be autogenerated from the yaml entries. What, after all, if CUDA is not installed? log_softmax_sparse_cuda won't be available.

Reply:
That is a good point, and that one of  the reason to use the device dispatcher if we want to avoid duplicate code. As I explained before, I tried to use in many ways the device dispatcher in the sparse module to reduce common code between cpu and cuda code, however it seems that it only works with specific paths. Look at this [TODO](https://github.com/pytorch/pytorch/blob/9d1138afec26a4fe0be74187e4064076f8d45de7/aten/src/ATen/native/DispatchStub.h#L35). 

So I think that in order to maintain compatibility even when CUDA is not enabled I will go back and leave the dispatch logic to be autogenerated from yaml entries. cc @pearu